libraries/AP_RangeFinder/AP_RangeFinder.cpp

```diff
@@ -857,7 +859,33 @@ bool RangeFinder::get_temp(enum Rotation orientation, float &temp) const
     }
     return backend->get_temp(temp);
 }
-
+// get temperature (with external sensor override support)
+bool RangeFinder::get_temp_with_override(enum Rotation orientation, float &temp) const
```

Comment:
I’m unsure on what’s preferable out of renaming the original function to this, or just implementing the override “silently” within the `get_temp` function (since as far as the using code is concerned they’re getting whatever temperature is relevant to the rangefinder).

Either way, I don’t think the internal function (which does not factor in the override) should remain public - that seems like asking for someone to use it without realising they’re supposed to be using the override one, which would lead to bugs where the override is inconsistently applied.